Standard tests: update template (#2937)
airbytehq/airbyte: docs/contributing-to-airbyte/building-new-connector/testing-connectors.md

### Setting up standard tests for your connector

> Standard tests are typically run from a docker container. 

Reviewer: I think we should remove all references to the v1 test suite into a separate md page (lets say maybe `Standard Tests (Legacy)`) and link to it in this page saying something like `If you are looking for reference documentation for the deprecated first version of test suites, see <link>`. 

The current md file has a little bit of split-brain between v1 and v2. I would rather we start fresh for a v2 document and feel free to copy anything from the v1 document that you like.

Since the inputs to standard tests are often static, the file-based runner is sufficient for most connectors. However, in some cases, you may need to run pre or post hooks to dynamically create or destroy resources for use in standard tests. For example, if we need to spin up a Redshift cluster to use in the test then tear it down afterwards, we need the ability to run code before and after the tests, as well as customize the Redshift cluster URL we pass to the standard tests. If you have need for this use case, please reach out to us via [Github](https://github.com/airbytehq/airbyte) or [Slack](https://slack.airbyte.io). We currently support it for Java & Python, and other languages can be made available upon request.

## Airbyte's Standard Tests (v2) - Source Acceptance Tests

Unfortunately Standard Tests (v1) has some problems: 
* Its tests are opaque to a contributor, difficult to debug when something goes wrong
* It tries to encompass testing all features/edge cases in one test case which has made it difficult to configure and its behavior hard to understand

Source Acceptance Tests tries to address these issues. These are the differences from v1:

> * configured with YAML file, forget about Gradle

Reviewer: ```suggestion
* configured with YAML file and run via Docker or pytest
```

> * individual tests can be disabled in the config file

Reviewer: ```suggestion
* test suites can be enabled/disabled via config 
```

Reviewer: I think `individual tests` makes it sound like test cases not test suites (e.g: incremental is a suite)

> * individual tests can have different inputs and run multiple time with different sets of inputs

Reviewer: ```suggestion
* test suites can run multiple time with different sets of inputs
```

* each test has a timeout

> * each test has stores all inputs and outputs 

Reviewer: ```suggestion
* each test stores all inputs and outputs, which can be used for debugging failures
```

> * use pytest as a test runner

Reviewer: ```suggestion
* uses pytest as a test runner
```

* can be run manually, with interactive debugger, custom plugins, last failed tests, etc - thanks to pytest

> * the output is easy to read, good bye Java stacktrace - thanks to pytest

Reviewer: Feels like this is "apologizing" for mistakes in the past but most people will not have used the previous one so we should focus on their experience

```suggestion
```
